components/inputs/input-number.js

```diff
@@ -438,7 +438,7 @@ class InputNumber extends FocusMixin(LabelledMixin(SkeletonMixin(FormElementMixi
 		}
 
 		if (dispatchEvent) {
-			await this.updateComplete; // wait for validity logic to re-run
+			await this.requestValidate(true); // wait for validity logic to re-run
```

Comment:
Wow... the validation stuff is really 🍌 s eh?!?

Does this mean that `requestValidate` gets called twice, once from `updated`, and once here? I see it's also called from `FormElementMixin`'s `updated` callback, but I'm not clear on when it would actually run.

Is it strange that `FormElementMixin`'s `updated` callback doesn't call `super.updated`? Maybe that was intentional... 🤔

Reply:
> Does this mean that requestValidate gets called twice, once from updated, and once here?

Yup, although this wouldn't be the first time -- input-text and input-textarea both call it twice as well, likely for similar reasons.

> Is it strange that FormElementMixin's updated callback doesn't call super.updated?

Hmm, yeah that is a little odd. Like the Lit docs say you don't need to, but I'm assuming that's only when you're extending `LitElement` directly...